src/workflows/airqo_etl_utils/satellite_data_utils.py

```diff
@@ -0,0 +1,229 @@
+from datetime import datetime, timedelta
+import datetime as dt
+from google.cloud import bigquery
+from google.oauth2 import service_account
+import pandas as pd
+import numpy as np
+import os
+import random
+import string
+import ee
+from functools import reduce
+from pymongo import MongoClient
+from bson.json_util import dumps
+
+from pymongo import MongoClient
+
+from .config import Config
+
+credentials = ee.ServiceAccountCredentials(
+    key_file=Config.GOOGLE_APPLICATION_CREDENTIALS,
+    email=Config.GOOGLE_APPLICATION_CREDENTIALS_EMAIL,
+)
+class DataHandler:
+    def __init__(self):
+        self.client = bigquery.Client()
+        self.data_path = None
+        ee.Initialize(credentials, project=Config.GOOGLE_CLOUD_PROJECT_ID)
+        self.mongo_client = MongoClient(Config.MONGODB_URI)
+        self.db = self.mongo_client[Config.MONGODB_DATABASE]
+        self.collection = self.db[Config.MONGODB_COLLECTION]
+
+    SEED = 89561
+
+    def seeder(self, seed):
+        random.seed(seed)
+        np.random.default_rng(seed)
+        np.random.seed(seed)
+
+    def random_id(self, N=10):
+        return ''.join(random.choice(string.ascii_lowercase + string.digits) for _ in range(N))
+    
+    def query_bigquery_batch(self, start_time=None, end_time=None, batch_size=500):
+        if not start_time:
+            start_time = datetime.now() - timedelta(days=7)
+        if not end_time:
+            end_time = datetime.now()
+
+        query = f"""
+            SELECT 
+                site_id, timestamp, site_name, site_latitude, site_longitude, pm2_5,
+                pm10, country
+            FROM {Config.BIGQUERY_ANALYTICS_TABLE}
+            WHERE timestamp BETWEEN TIMESTAMP('{start_time.isoformat()}')
+                AND TIMESTAMP('{end_time.isoformat()}')
+                AND pm2_5 IS NOT NULL
+                AND site_id IS NOT NULL
+                AND site_latitude IS NOT NULL
+            LIMIT {batch_size};
+        """
+
+        try:
+            query_job = self.client.query(query)
+            df = query_job.to_dataframe() 
+            df = df.sort_values(by=['site_name', 'timestamp']).reset_index(drop=True)
+            df['date'] = [str(x.date()) for x in df.timestamp]
+            df['month'] = df.timestamp.dt.month
+            df['hour'] = df.timestamp.dt.hour
+            return df
+        except Exception as e:
+            print(f"Error querying BigQuery: {e}")
+            return None
```

Comment:
@wabinyai most of these bigquery related methods are in `bigquery_api.py` so you could move this there